### PR TITLE
DCS-640 Tidying up application creation

### PR DESCRIPTION
**Pull request by andrewrlee**
Separating out a service index.ts which does the service wiring up from
    the creation of the app.
    
    The type for the services container can then be passed straight through to the routers
    rather than threading through each service independently.
    
    This had a knock on effect of forcing tests to be tidied up a bit

Also moving reporting client over to class.

### CODE REVIEW

File: server/routes/coordinator.test.ts [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
+
+jest.mock('../services/offenderService')
+jest.mock('../services/report/reportService')
+jest.mock('../services/involvedStaffService')
+jest.mock('../services/reviewService')
+
+const offenderService = new OffenderService(null) as jest.Mocked<OffenderService>
+const reportService = new ReportService(null, null, null) as jest.Mocked<ReportService>
+const involvedStaffService = new InvolvedStaffService(null, null, null, null) as jest.Mocked<InvolvedStaffService>
+const reviewService = new ReviewService(null, null, null, null, null) as jest.Mocked<ReviewService>
```
Comment (andrewrlee) [inline server/routes/coordinator.test.ts:15]: now using jest automatic mocks

File: server/routes/coordinator.test.ts [hunk 2]
```diff
@@ -1,23 +1,21 @@
-const request = require('supertest')
-const { appWithAllRoutes, user, reviewerUser, coordinatorUser } = require('./testutils/appSetup')
+import request from 'supertest'
+import { StaffDetails } from '../data/draftReportClientTypes'
+import { InvolvedStaffService, OffenderService, ReportService, ReviewService } from '../services'
```
Comment (andrewrlee) [inline server/routes/coordinator.test.ts:3]: all services being re-exported from one place reduces lines of imports

File: server/index.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { services } from './services'
+import createApp from './app'
+
+const app = createApp(services)
+
+export default app
```
Comment (andrewrlee) [inline server/index.ts:6]: previously creating object graph and app creation happened in the same place now separated out

File: server/app.ts [hunk 4]
```diff
@@ -235,37 +225,13 @@ export default function createApp({
...
-    '/api/',
-    createApiRouter({
-      authenticationMiddleware,
-      offenderService,
-      reportingService,
-      systemToken,
-    })
-  )
+  app.use('/', createRouter(authenticationMiddleware, services))
+  app.use('/api/', createApiRouter(authenticationMiddleware, services))
```
Comment (andrewrlee) [inline server/app.ts:234]: passing services straight through now

File: server.js [hunk 5]
```diff
@@ -8,8 +8,7 @@ initialiseAppInsights()
 
 const knex = require('knex')
 const knexfile = require('./knexfile')
-/** @type {any} */
-const app = require('./server/index')
+const app = require('./server/index').default
```
Comment (andrewrlee) [inline server.js:11]: typescript now knows that `app` is of type `Express`

File: server/services/index.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  prisonerSearchService,
+  reviewService,
+  reportingService,
+  systemToken,
+  locationService,
+  reportDetailBuilder,
+  draftReportService,
+}
+
+export type Services = typeof services
```
Comment (andrewrlee) [inline server/services/index.ts:96]: export type of the services data structure as a type in its own right

File: server/services/index.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+  DraftReportService,
+  ReportService,
+  ReportingService,
+  LocationService,
+  StatementService,
+  ReviewService,
+  OffenderService,
+  PrisonerSearchService,
+  ReportDetailBuilder,
+}
```
Comment (andrewrlee) [inline server/services/index.ts:109]: re-export these to provide central place to import from.

File: server/services/involvedStaffService.ts [hunk 8]
```diff
@@ -81,7 +81,7 @@ export class InvolvedStaffService {
     })
   }
 
-  public removeInvolvedStaff = async (reportId: number, statementId: number): Promise<void> => {
+  public async removeInvolvedStaff(reportId: number, statementId: number): Promise<void> {
```
Comment (andrewrlee) [inline server/services/involvedStaffService.ts:84]: jest auto mocking doesn't work for class properties that are functions - only methods

File: server/types/uof.d.ts [hunk 9]
```diff
@@ -35,18 +35,6 @@ export type GetUsersResults = {
...
-  getMostOftenInvolvedStaff: (agencyId: AgencyId, range: DateRange) => Promise<Array<any>>
-  getMostOftenInvolvedPrisoners: (agencyId: AgencyId, range: DateRange) => Promise<Array<any>>
-  getIncidentsOverview: (agencyId: AgencyId, range: DateRange, statuses) => Promise<Array<any>>
-  getIncidentLocationsAndTimes: (agencyId: AgencyId, range: DateRange) => Promise<Array<any>>
-  getIncidentCountByOffenderNo: (agencyId: AgencyId, range: DateRange) => Promise<Array<OffenderNoWithIncidentCount>>
-  getIncidentsForAgencyAndDateRange: (
-    agencyId: AgencyId,
-    range: DateRange
-  ) => Promise<Array<OffenderNoWithIncidentDate>>
-}
```
Comment (andrewrlee) [inline server/types/uof.d.ts:48]: get rid of unnecessary type def as class exports its own type

**Review by bell-pa — APPROVED** [review]
LGTM

**Pull request merged**